### ISSUE TITLE
[Shaman][Spell] Double Windfury Weapon proccrate too low

### ISSUE THREAD

**Issue by quapuzi** [Atlantiss/NetherwingBugtracker#6209]
**Description**: 
The Shamans Windfury Weapon enchant should interact with each other. If you use Windfury on each weapon, the proccrate should go from 20% to 36% for each Weapon.

**Current behaviour**:
Looking at logs from Shamans that use double windfury weapon, you see no increase of the proccchance. Proof:
https://legacyplayers.com/Raids/Viewer/Default.aspx?id=8321&exp=1&upl=12473
Taking the Mother kill, Lillefood has 362 Autohits landing, plus 46x2 Stormstrike hits.
Thats 454 Hits that can procc Windfury.
Now looking at his actual Windfury hits, you see 102 Offhand hits and 50 Mainhand hits.
Since Windfury Weapon always does 2 Hits, that is (102+50)/2 = 76 Windfury proccs.
That means, the wf procc chance is (76/454)*100= 16.7%
Technically you could say this is right, since Windfury Weapon has an internal cooldown, but 2 Things are super weird:
First the procchance on the offhand is higher. Statistically it should be the same or almost the same. This makes no sense. Also a slow weapon causes less windfury proccs, and hes using a slower weapon (Siphon of the Nathrazim, 2.8 speed vs Planar Edge, 2.7 speed).
Proof for slow weapons causing less proccs:
https://web.archive.org/web/20080811103051/http://elitistjerks.com/f47/t20765-shaman_enhancement/#post846190
Post from ElginRoko:
Using a 2.6 and a 2.8 as stressed earlier, the model shows the 2.6 speed weapon proccing around 12% more often, whilst being only 7.7% faster.
This means, the faster Mainhand should procc even more often.

Now looking at a log of a Shaman using Windfury/Flametongue with 2 Slow Weapons (2.7 and 2.6):
https://legacyplayers.com/Raids/Viewer/Default.aspx?id=8396&exp=1
Ezek has 729 Autohits that land. Assuming that its 50/50 for Mainhand and Offhand, youre at
729/2=~364 Hits
He did also 83x Stormstrike, wich means 364+83=447 Possible Wf proccs
Looking at Windfury, he has 134 Hits. Again, since Windfury weapon hits twice, thats
134/2= 67 Windfury proccs.
So his actual Windfury chance is:
(67/447)*100= 14.99%
This is about the same as the one from Lillefood, and you can include some delta because of RNG, flurry uptime, downtime etc. So technically this seems right. Also 15% - 16% would be right, if Windfury weapon has 20% proccchance and youre not proccing on CD. Not Matter if you use Windfury weapon once or twice. The only odd thing is, that the offhand is proccing way more often than the Mainhand.

**Expected behaviour**: 
Using Windfury on both Weapons should increase your procc chance from 20% to 36% per Weapon. Proof:
https://web.archive.org/web/20080811214521/http://elitistjerks.com/f31/t22760-enhancement_shaman_yo_help_me/

Post of • Toots Hepcat:
---------------------------------------------------------------------------------------------------

The reason 2x WF is superior to 1x WF, even though some of your WF attacks are smaller offhand attacks, is simple: when you have WF on one hand, the chance for that one hand to swing is 20%. However, when you put WF on BOTH hands, EACH hand has its proc chance increased to 36%.

So in essence, you're not getting twice the chance to proc WF, you're getting about three times the chance.

What's more, no matter how many hands you put WF on, it now has a 3s cooldown in between procs. That means you'll never hit a 20% proc rate with any weapon slower than 3s -- the first attack after a windfury has a 0% chance to proc. Thus, this tripling of your chance to proc WF actually improves your damage from MH WF procs, too! (The Main Hand has a natural advantage to proccing Windfury, regardless of weapon speed, because Stormstrike always evaluates the main hand first).

Neither of these divergences from the tooltip -- the 3s CD and the 36% chance to proc on either hand -- are documented publicly by Blizzard, but either is easily detectable in any combat log. They've been around since patch 2.1, which I believe is a year old.

You can see this first hand without doing any theory. Get a copy of Recount, then go farm with WF MH and FT OH, then farm with 2x WF on some slow weapons. I guarantee you will see more DPS. And as your gear improves, WF just keeps getting better and better.
---------------------------------------------------------------------------------------------------

First thing is, that even excluding the higher procc chance, Windfury weapon should hit more often with your Mainhand than your Offhand, because: " (The Main Hand has a natural advantage to proccing Windfury, regardless of weapon speed, because Stormstrike always evaluates the main hand first)."
This is clearly not the case on NW, its the opposite.

Also hes saying, that there was no bluepost from Blizzard about the increase from 20% to 36%:
"Neither of these divergences from the tooltip -- the 3s CD and the 36% chance to proc on either hand -- are documented publicly by Blizzard,"
This post is from 04/18/08, means its from patch 2.4.1 (3rd last patch). And very late into tbc, when Wotlk was already getting developed with completely different mechanics. So finding any bluepost about this is probably impossible.
Also hes saying that youll see it by just trying it impiric, wich obviously is not possible anymore.

Another evidence for this:
https://web.archive.org/web/20080811103051/http://elitistjerks.com/f47/t20765-shaman_enhancement/#post846190
Post from Atren:
When you dual wield weapons, the chance on each landed attack outside the 3-second cooldown is approximatedly 36%, if and only if both weapons are imbued with Windfury. [footnote]Originally posted by Disquette on the WoW official forums. Original post is no longer archived by Blizzard.[/footnote]

Hes also saying that this post is no longer available as of May 2008.

All these numbers are based on Wowwebstats, a site that does not exist anymore since late Wotlk.
 Of course there can be RNG, and since Windfury Weapon has a ICD, it wont be landing on 36% off all hits. But the same guy is also saying:
"Analysis of the combat log shows that if you sum all hits, the proc rate while DWing is 20%, but that includes hits you make while inside the 3 second cooldown, which cannot actually proc WF. When you remove the ineligible hits the observed proc rate from the eligible hits becomes 36%."

This means, the actual ratio of Autohits+Stormstrike while having 2X Windfury Weapon should be
20%, while in reality it is 16%. 16% is most likely right for 1x Windfury Weapon tho.

TLDR: Windfury Weapon on one weapon should have 20% Proccchance, resulting in 16% procc rate in reality.
Windfury Weapon on both hands should have a 36% Proccchance per Weapon, resulting in 20% procc rate. Also Windfury procc caused by Stormstrike should always prefer the Mainhand, since the Mainhand is always executed first (no proof if it does or not now, i dont have the code from the server).
Also from a pure statistic standpoint, the mainhand should procc as often as the offhand, if not even more.

**Server Revision**: 
3101

**Comment by Wolffenstein** [comment]
Fixed in rev 3104.
